tests/e2e/specs/backend/product-query/atomic-blocks.test.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { canvas, setPostContent, insertBlock } from '@wordpress/e2e-test-utils';
+import { visitBlockPage, saveOrPublish } from '@woocommerce/blocks-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { GUTENBERG_EDITOR_CONTEXT, describeOrSkip } from '../../../utils';
+import { shopper } from '../../../../utils';
+
+const block = {
+	name: 'Product Query',
+	slug: 'woocommerce/product-query',
+	class: '.wp-block-query',
+};
+
+describeOrSkip( GUTENBERG_EDITOR_CONTEXT === 'gutenberg' )(
+	'Product Query > Atomic blocks',
+	() => {
+		beforeEach( async () => {
+			await visitBlockPage( `${ block.name } Block` );
```

Comment:
I suggest you add `waitForCanvas()` here in the `beforeEach`, it might not seem necessary, but since you are using `canvas()` in your test cases, believe me, it's going to eliminate some false negatives and add some safety.

Reply:
FYI, updated `waitForCanvas()` to support Page Editor. Gutenberg hasn't migrated Page Editor to use iframe yet.